mpitests/test_expval.py

```diff
@@ -22,114 +22,265 @@
+            qml.Hadamard,
+            pytest.param(
+                qml.Identity,
+                marks=pytest.mark.xfail(
+                    reason="The Identity gate need a deep review for MPI support"
+                ),
+            ),
+        ],
+    )
+    @pytest.mark.parametrize("wires", [1, 2, numQubits - 2, numQubits - 1, numQubits])
+    def test_expval_single_wire_no_parameters(self, tol, operation, wires, C_DTYPE, batch_obs):
+        """Tests that expectation values are properly calculated for single-wire observables without parameters."""
+        num_wires = wires
+        comm = MPI.COMM_WORLD
+
+        dev_mpi = qml.device(
+            "lightning.gpu", wires=numQubits, mpi=True, c_dtype=C_DTYPE, batch_obs=batch_obs
         )
 
-    def test_pauliy_expectation(self, theta, phi, tol):
-        """Test that PauliY expectation value is correct"""
-        dev = qml.device(device_name, mpi=True, wires=3)
+        dev_cpu = qml.device("lightning.qubit", wires=num_wires, c_dtype=C_DTYPE)
+
+        state_vector = create_random_init_state(num_wires, dev_mpi.c_dtype)
+        comm.Bcast(state_vector, root=0)
+
+        def circuit():
+            qml.StatePrep(state_vector, wires=range(num_wires))
+            return qml.expval(operation(0))
+
+        cpu_qnode = qml.QNode(circuit, dev_cpu)
+        expected_output_cpu = cpu_qnode()
+        comm.Bcast(np.array(expected_output_cpu), root=0)
+
+        mpi_qnode = qml.QNode(circuit, dev_mpi)
+        expected_output_mpi = mpi_qnode()
+
+        assert np.allclose(expected_output_mpi, expected_output_cpu, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize(
+        "obs",
+        [
+            qml.PauliX(0) @ qml.PauliZ(1),
+            qml.PauliX(0) @ qml.PauliZ(numQubits - 1),
+            qml.PauliX(numQubits - 2) @ qml.PauliZ(numQubits - 1),
+            qml.PauliZ(0) @ qml.PauliZ(1),
+            qml.PauliZ(0) @ qml.PauliZ(numQubits - 1),
+            qml.PauliZ(numQubits - 2) @ qml.PauliZ(numQubits - 1),
+        ],
+    )
+    def test_expval_multiple_obs(self, obs, tol, C_DTYPE, batch_obs):
+        """Test expval with Hamiltonian"""
+        num_wires = numQubits
+
+        dev_cpu = qml.device("lightning.qubit", wires=num_wires, c_dtype=C_DTYPE)
+        dev_mpi = qml.device(
+            "lightning.gpu", wires=num_wires, mpi=True, c_dtype=C_DTYPE, batch_obs=batch_obs
+        )
 
-        O1 = qml.PauliY(wires=[0])
-        O2 = qml.PauliY(wires=[1])
+        def circuit():
+            qml.RX(0.4, wires=[0])
+            qml.RY(-0.2, wires=[num_wires - 1])
+            return qml.expval(obs)
 
-        dev.apply(
-            [qml.RX(theta, wires=[0]), qml.RX(phi, wires=[1]), qml.CNOT(wires=[0, 1])],
-            rotations=[*O1.diagonalizing_gates(), *O2.diagonalizing_gates()],
+        cpu_qnode = qml.QNode(circuit, dev_cpu)
+        mpi_qnode = qml.QNode(circuit, dev_mpi)
+
+        assert np.allclose(cpu_qnode(), mpi_qnode(), atol=tol, rtol=0)
```

Comment:
For results we already know the answer to, should we be able to avoid running an additional CPU-side simulation by pre-defining the values we anticipate?

Same with all uses of `cpu_qnode` below --- we should only need a heavy system-level test if we cannot determine the results beforehand. Additionally, if LQ and LG break in the same way, we cannot know if there is a mistake.

What are your thoughts?

Reply:
Yes, you are right. However, the purpose of these tests is to check if LG-MPI reproduces the same result as LQ. The correctness of LQ is tested in the other test suite (dir `tests`).  Although I can move the evaluation to something more analytical but it will require more time, perhaps we can define a new story in SC. 
What do you think? :smile: